src/com/xilinx/rapidwright/edif/EDIFNetlist.java

```diff
@@ -1790,6 +1791,88 @@ public Map<EDIFCell, List<EDIFChange>> getModifiedCells() {
         return modifiedCells;
     }
 
+    /**
+     * Helper method to get the IOStandard property of an EDIFCellInst;
+     * should one not exist, fallback to the IOStandard property on
+     * the EDIFNet object connected to the top-level port that it drives.
+     * On the first time a fallback is considered, a map is initialized
+     * {@link #cellInstIOStandardFallback} to speed up future queries.
+     * This map will not be updated upon connecting a new EDIFNet nor
+     * if a new property was added to the existing EDIFNet --- for this
+     * map to be re-initialized use {@link #resetCellInstIOStandardFallbackMap}.
+     * @param eci EDIFCellInst object.
+     * @return EDIFPropertyValue describing its IOStandard. Returns
+     *         EDIFNetlist.DEFAULT_PROP_VALUE if no value found.
+     */
+    public EDIFPropertyValue getIOStandard(EDIFCellInst eci) {
+        EDIFPropertyValue value = eci.getIOStandard();
+        if (value != DEFAULT_PROP_VALUE) {
+            return value;
+        }
+
+        if (cellInstIOStandardFallback == null) {
+            cellInstIOStandardFallback = new HashMap<>();
+
+            EDIFHierCellInst topEhci = getTopHierCellInst();
+            for (EDIFPort ep : getTopCell().getPorts()) {
+                if (ep.isInput()) {
+                    // Only handle output and inout ports
```

Comment:
How did you conclude that inputs cannot have properties propagated back?

Reply:
1. I haven't seen a need for IOStandard propagation for input buffers, which was especially necessary for transforming output buffers.
2. Nets for input buffers would need to be propagated forward, rather than backwards. Let me leave a comment behind.